rama-dns/src/lib.rs

```diff
@@ -79,6 +80,37 @@ impl<R: DnsResolver<Error: Into<BoxError>>> DnsResolver for Option<R> {
     }
 }
 
+static_str_error! {
+    #[doc = "domain not mapped in any dns resolver"]
+    pub struct DomainNotFoundErr;
```

Comment:
I think it's fine to have a custom error for it, but I would prefer if you implement a custom one by hand so we can keep track of _all_ errors:

```
pub struct DnsChainDomainResolveErr<E> {
    errors: Vec<E>,
}

impl<E: fmt::Debug> fmt::Debug for DnsChainDomainResolveErr<E> { ... }

impl<E: std::error::Error> std::error::Error for DnsChainDomainResolveErr<E> {
   fn source(...) { ... /* return last error as some */ }
}
```

This way you can keep track of errors with:

```rust
let mut errors = Vec::new();

// ...

Err(err) => errors.push(err),
```